### PR TITLE
Escape HTML in media title and creator

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes WordPress/openverse-infrastructure#593 by @sarayourfriend 

## Description

This PR escapes the media title and creator in the attribution text to prevent script injection. It uses `String.replace` because the [HTML Sanitizer API](https://developer.mozilla.org/en-US/docs/Web/API/HTML_Sanitizer_API) is still experimental.

## Testing Instructions

You should not be able to reproduce the bug outlined in the linked issue.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [ ] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/test/unit/specs/utils/attribution-html.spec.ts [hunk 1]
```diff
@@ -62,6 +62,17 @@ describe("getAttribution", () => {
...
 
+  it("escapes embedded HTML", () => {
+    const mediaItemWithHtml = {
+      ...mediaItem,
+      originalTitle: '<script>console.log("HELLO");</script>',
+    }
+    const attrText = getAttribution(mediaItemWithHtml, i18n)
+    const expectation =
+      "&lt;script&gt;console.log(&quot;HELLO&quot;);&lt;/script&gt;"
+    expect(attrText).toContain(expectation)
```
Comment (sarayourfriend) [inline frontend/test/unit/specs/utils/attribution-html.spec.ts:73]: Maybe also this, for good measure?

```suggestion
    expect(attrText).toContain(expectation)
    expect(attrText).not.toContain(mediaItemWithHtml.originalTitle)
```

**Review by sarayourfriend — APPROVED** [review]
Nice! Left one suggestion for an additional assertion but LGTM otherwise.

**Review by obulat — COMMENTED** [review]
This works, but wouldn't it be better to move the escaping to `decode-media-data`:

https://github.com/WordPress/openverse/blob/0cb72d30254f4d1f54497320018288c1373520b8/frontend/src/utils/decode-media-data.ts#L70

https://github.com/WordPress/openverse/blob/0cb72d30254f4d1f54497320018288c1373520b8/frontend/src/utils/decode-media-data.ts#L100

**Comment by dhruvkb** [comment]
That was my first instinct @obulat but that leads to double escaping issues where `v-html` is not used (such as the title).

Screenshot 2023-08-22 at 2 05 38 PM

**Review by obulat — APPROVED** [review]

**Pull request merged**